sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -351,4 +353,27 @@ public static Duration getDefaultTimeoutFromEnvironment(Configuration configurat
             return defaultTimeout;
         }
     }
+
+    /**
+     * Merges two {@link Context Contexts} into a new {@link Context}.
+     *
+     * @param into Context being merged into.
+     * @param from Context being merged.
+     * @return A new Context that is the merged Contexts.
+     * @throws NullPointerException If either {@code into} or {@code from} is null.
+     */
+    public static Context mergeContexts(Context into, Context from) {
+        Objects.requireNonNull(into, "'into' cannot be null.");
+        Objects.requireNonNull(from, "'from' cannot be null.");
+
+        Stack<Context> fromContextStack = from.getValueStack();
+
+        Context returnContext = into;
+        while (!fromContextStack.empty()) {
+            Context toAdd = fromContextStack.pop();
+            returnContext = returnContext.addData(toAdd.getKey(), toAdd.getValue());
+        }
```

Comment:
Couldn't we avoid creating the temporary stack if we moved the merge operation into the `Context` class itself?

Reply:
Possibly, the issue is that Context is represented in a single-direction linked list with the newest Context first, but when merging the oldest should be the first one merged